packages/core/src/sink/IndexSink.ts

```diff
@@ -0,0 +1,41 @@
+/** @license MIT License (c) copyright 2010-2016 original author or authors */
+/** @author Brian Cavalier */
+/** @author John Hann */
+
+import Pipe from './Pipe'
+import { Time, Sink } from '@most/types'
+
+export interface IndexedValue<A> {
+  readonly index: number
+  readonly value: A
+  readonly active: boolean
+}
+
+export default class IndexSink<A> extends Pipe<A | Readonly<IndexedValue<A | undefined>>> implements Sink<A | Readonly<IndexedValue<A | undefined>>> {
```

Comment:
This seems like it should just `extends Pipe<A>` since it only _accepts_ `A`s.  Can you help me understand why it needs to be `extends Pipe<A | Readonly<IndexedValue<...>>>`?

Reply:
see https://github.com/mostjs/core/pull/296#discussion_r316125789